onnxruntime/core/providers/mkldnn/subgraph/MklBatchNorm.h

```diff
@@ -0,0 +1,308 @@
+    constexpr int kNumInputScaleDimensions = 1;
+    constexpr int kNumInputBiasDimensions = 1;
+    constexpr int kNumInputMeanDimensions = 1;
+    constexpr int kNumInputVarianceDimensions = 1;
+
+    if (xshape.GetDims().empty()) {
+      return common::Status(common::ONNXRUNTIME, common::INVALID_ARGUMENT, "Invalid input X: Empty dimensions");
+    }
+
+    int64_t num_channels = xshape.GetDims()[1];
+
+    if (scale_shape.NumDimensions() != kNumInputScaleDimensions) {
+      return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "Invalid input scale: NumDimensions() != ", kNumInputScaleDimensions);
+    }
+    if (scale_shape.GetDims()[0] != num_channels) {
+      return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "Invalid input scale: 0th dimension != ", num_channels);
+    }
+
+    if (b_shape.NumDimensions() != kNumInputBiasDimensions) {
+      return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "Invalid input B: NumDimensions() != ", kNumInputBiasDimensions);
+    }
+    if (b_shape.GetDims()[0] != num_channels) {
+      return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "Invalid input B: 0th dimension != ", num_channels);
+    }
+
+    if (mean_shape.NumDimensions() != kNumInputMeanDimensions) {
+      return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "Invalid input mean: NumDimensions() != ", kNumInputMeanDimensions);
+    }
+    if (mean_shape.GetDims()[0] != num_channels) {
+      return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "Invalid input mean: 0th dimension != ", num_channels);
+    }
+
+    if (var_shape.NumDimensions() != kNumInputVarianceDimensions) {
+      return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "Invalid input var: NumDimensions() != ", kNumInputVarianceDimensions);
+    }
+    if (var_shape.GetDims()[0] != num_channels) {
+      return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "Invalid input var: 0th dimension != ", num_channels);
+    }
+    return common::Status::OK();
+  }
+
+  static void NormalizeDims(const TensorShape& x_shape, std::vector<int64_t>& new_dims) {
+    new_dims.clear();
+    auto& orig_dims = x_shape.GetDims();
+    if (orig_dims.size() == 4 /*supported size by CUDA*/ ||
+        orig_dims.size() == 5 /*supported size by CUDA*/) {
+      new_dims = orig_dims;
+      return;
+    }
+
+    auto rank = x_shape.NumDimensions();
+    auto num_samples = rank > 0 ? orig_dims[0] : 1;  // NCHW
+    auto num_channels = rank > 1 ? orig_dims[1] : 1;
+    auto width = rank > 3 ? orig_dims[3] : 1;
+    auto height = rank > 2 ? orig_dims[2] : 1;
+    new_dims = {num_samples, num_channels, height, width};
+  }
+};
+
+template <typename T>
+class MklBatchNorm : public MklKernel {
+ public:
+  explicit MklBatchNorm(MklNode& node,
+                        MKLDNNExecutionProvider* provider,
+                        std::shared_ptr<MKLContext> mkl_context) : MklKernel(node, provider, mkl_context) {
+  }
+  void ReadAttributes(const std::unordered_map<std::string,
```

Comment:
unordered_map<std::string, ONNX_NAMESPACE::AttributeProto> is  just NodeAttributes.
seems to be inconsistently used with your changes. 
please replace all occurrences